agent/api/task/service_connect.go

```diff
@@ -13,13 +13,29 @@
 
 package task
 
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/aws/amazon-ecs-agent/agent/acs/model/ecsacs"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+const (
+	serviceConnectConfigKey        = "ServiceConnectConfig"
+	serviceConnectContainerNameKey = "ContainerName"
+)
+
 // ServiceConnectConfig represents the Service Connect configuration for a task.
 type ServiceConnectConfig struct {
 	ContainerName string               `json:"containerName"`
 	StatsConfig   StatsConfig          `json:"statsConfig"`
-	IngressConfig []IngressConfigEntry `json:"ingressConfig"`
-	EgressConfig  EgressConfig         `json:"egressConfig"`
-	DNSConfig     []DNSConfigEntry     `json:"dnsConfig"`
+	IngressConfig []IngressConfigEntry `json:"ingressConfig,omitempty"`
+	EgressConfig  EgressConfig         `json:"egressConfig,omitempty"`
```

Comment:
`omitempty` doesnt really help here since the values inside the struct would still be initialized. It is better to use pointers here. Same applies to `EgressConfig.ListenerPort` and `VIP.IPV4CIDR` and `VIP.IPV6CIDR` below. [This](https://www.sohamkamani.com/golang/omitempty/) has a good explanation of `omitempty`

Reply:
Types are updated in commit 3d8d95406e74fc25d3c00d384a11cd5cd4ae55f9. Please take a look on `service_connect.go`. Thanks.